tests/cnf/core/network/metallb/tests/bgp-remote-as-dynamic.go

```diff
@@ -0,0 +1,215 @@
+
+					By("Checking that BGP session is established and up")
+					verifyMetalLbBGPSessionsAreUPOnFrrPod(frrPod, removePrefixFromIPList(ipv4NodeAddrList))
+
+					By("Validating external FRR AS number received on the FRR nodes")
+					Eventually(func() error {
+						return frr.ValidateBGPRemoteAS(frrk8sPods, ipv4metalLbIPList[0], tsparams.RemoteBGPASN)
+					}, 60*time.Second, 5*time.Second).Should(Succeed(),
+						fmt.Sprintf("The remoteASN does not match the expected AS: %d", tsparams.RemoteBGPASN))
+				})
+
+			It("Verify the establishment of an iBGP adjacency using neighbor peer remote-as internal",
+				reportxml.ID("76822"), func() {
+					By("Step up test cases with Frr Node AS 64500 and external Frr AS 64500")
+					frrk8sPods, frrPod := setupBGPRemoteASTestCase(addressPool, hubIPv4ExternalAddresses, externalAdvertisedIPv4Routes,
+						externalAdvertisedIPv6Routes, dynamicASiBGP, tsparams.LocalBGPASN)
+
+					By("Checking that BGP session is established and up")
+					verifyMetalLbBGPSessionsAreUPOnFrrPod(frrPod, removePrefixFromIPList(ipv4NodeAddrList))
+
+					By("Validating external FRR AS number received on the FRR nodes")
+					Eventually(func() error {
+						return frr.ValidateBGPRemoteAS(frrk8sPods, ipv4metalLbIPList[0], tsparams.LocalBGPASN)
+					}, 60*time.Second, 5*time.Second).Should(Succeed(),
+						fmt.Sprintf("The remoteASN does not match the expected AS: %d", tsparams.LocalBGPASN))
+				})
+
+			It("Verify the failure to establish a iBGP adjacency with a misconfigured external FRR pod",
+				reportxml.ID("76825"), func() {
+					By("Step up test cases with Frr Node AS 64500 and misconfigured iBGP external Frr AS 64501")
+					frrk8sPods, frrPod := setupBGPRemoteASTestCase(addressPool, hubIPv4ExternalAddresses, externalAdvertisedIPv4Routes,
+						externalAdvertisedIPv6Routes, dynamicASiBGP, tsparams.RemoteBGPASN)
+
+					By("Checking that BGP session is down")
+					verifyMetalLbBGPSessionsAreDownOnFrrPod(frrPod, removePrefixFromIPList(ipv4NodeAddrList))
+
+					By("Validating external FRR AS number received is incorrect and marked as 0 on the FRR nodes")
+					Eventually(func() error {
+						return frr.ValidateBGPRemoteAS(frrk8sPods, ipv4metalLbIPList[0], 0)
+					}, 60*time.Second, 5*time.Second).Should(Succeed(),
+						fmt.Sprintf("The remoteASN does not match the expected AS: %d", 0))
+				})
+		})
+	})
+
+func createBGPPeerWithDynamicASN(peerIP, dynamicASN string, eBgpMultiHop bool) {
+	By("Creating BGP Peer")
+
+	bgpPeer := metallb.NewBPGPeerBuilder(APIClient, "testpeer", NetConfig.MlbOperatorNamespace,
+		peerIP, tsparams.LocalBGPASN, 0).WithDynamicASN(mlbtypesv1beta2.DynamicASNMode(dynamicASN)).
+		WithPassword(tsparams.BGPPassword).WithEBGPMultiHop(eBgpMultiHop)
+
+	_, err := bgpPeer.Create()
+	Expect(err).ToNot(HaveOccurred(), "Failed to create BGP peer")
+}
+
+func deployFrrExternalPod(addressPool, hubIPAddresses, externalAdvertisedIPv4Routes,
```

Comment:
it's not only deployFrrExternalPod. It creates Bgpadv, ipaddresspool, nginx pod ...